### DISCUSSION TITLE
Stork for Choosing the Application Bean

### DISCUSSION THREAD

**Discussion by akil-rails** [quarkusio/quarkus#42674] [Q&A] [answered]
Hi

I have a use case, where I have to choose one of the available application beans basis one either the response time or simple round robin. Requests come to the app's rest resource , and the rest resource has to choose one the these services basis the strategy that's configured (the objective is for testing alternative implications of a service interface - like A/B testing for clients ) 

Is it possible , and is this a good use case for stork ? There are no network services involved - only application beans ( they may internally have some network calls to a database or elsewhere ) but the load balancer is only limited to choosing the service .

The stork documentation talks about using this for network services via the rest client - hence this question.

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply, marked as answer]
You may be able to do this with the Stork API, but it was not one of the use cases we had in mind when we designed Stork (mostly about remote interaction and remote service discovery. 

However, if you implement your own service discovery (it's an SPI) to list your components and if all your components have the same "API", it should be doable.

Here are a few links:

- https://smallrye.io/smallrye-stork/latest/programmatic-api/
- https://smallrye.io/smallrye-stork/latest/service-discovery/custom-service-discovery/
